compiler/luci/pass/src/ConvertNCHWToNHWCPass.test.cpp

```diff
@@ -75,19 +76,44 @@ class SimpleGraph
 
     graph_input->dtype(loco::DataType::FLOAT32);
     input->dtype(loco::DataType::FLOAT32);
-    add->dtype(loco::DataType::FLOAT32);
-    beta->dtype(loco::DataType::FLOAT32);
     output->dtype(loco::DataType::FLOAT32);
     graph_output->dtype(loco::DataType::FLOAT32);
 
     uint32_t channel_size = 16;
     graph_input->shape({1, channel_size, 4, 4});
     input->shape({1, channel_size, 4, 4});
-    add->shape({1, channel_size, 4, 4});
-    beta->shape({1, channel_size, 1, 1});
     output->shape({1, channel_size, 4, 4});
     graph_output->shape({1, channel_size, 4, 4});
 
+    auto graph_body = insertGraphBody(input);
+    output->from(graph_body);
+  }
+
+  virtual ~SimpleGraph() = default;
+
+protected:
+  virtual loco::Node *insertGraphBody(loco::Node *input) = 0;
+
+public:
+  loco::Graph g;
+  luci::CircleInput *input = nullptr;
+  luci::CircleOutput *output = nullptr;
+};
+
+class AddGraph final : public SimpleGraph
+{
+  loco::Node *insertGraphBody(loco::Node *input) override
```

Comment:
default visibility is `private` if I'm correct. `insertGraphBody()`  is `protected` in base class.
is this your intention?

Reply:
insertGraphBody's visibility is protected as AddGraph inherits SimpleGraph in public.

My intention was to prohibit calling `insertGraphBody` in instances. I used protected just because it is a looser condition than private (using private is also ok in this test case).

Reply:
I mean, scope is different. base `SimpleGraph` is declared as `protected` and `AddGraph` used as default (seems `private`). 
My suggestion is like this to be specific.
```cxx
class AddGraph final : public SimpleGraph
{
protected:
  loco::Node *insertGraphBody(loco::Node *input) override
```

Reply:
Oh, I see. It looks better. I've updated the code. Thanks.